January 2024 release notes (#1342)
semgrep/semgrep-docs: docs/release-notes/january-2024.md

---
slug: january-2024
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: January 2024
tags:
  - Release notes
---

# Semgrep release notes for January 2024

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in January 2024:
  * [<i class="fas fa-external-link fa-xs"></i>1.55.1](https://github.com/semgrep/semgrep/releases/tag/v1.55.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.55.2](https://github.com/semgrep/semgrep/releases/tag/v1.55.2)
  * [<i class="fas fa-external-link fa-xs"></i>1.56.0](https://github.com/semgrep/semgrep/releases/tag/v1.56.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.57.0](https://github.com/semgrep/semgrep/releases/tag/v1.57.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.58.0](https://github.com/semgrep/semgrep/releases/tag/v1.58.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.59.0](https://github.com/semgrep/semgrep/releases/tag/v1.59.0)

## 🌐 Cloud Platform

### Added

> * Semgrep's Visual Studio Code extension now runs natively on Windows machines.

Author: @s-santillan This is now the only line in present tense. Do you have any thoughts on rewording? Maybe something like "Improved Semgrep's Visual Studio Code extension so that it runs natively on Windows machines" but not sure if that sounds off

Reviewer: Honestly, if we were a bigger enterprise team I'm sure we'd have a strictly enforced spec by now, but the only reason I flagged 56-60 was because they were partial sentences.

There's a case for both present and past tense for release notes. I historically followed Added/Changed/Removed (I wasn't the owner before).

> * Added ability for organizations to test connections to source code managers by going to

Reviewer: Last question about this - does this mean all 4 of our "officially" supported SCMs (GitHub, GitLab, Azure, BB?) - I don't think we need to spell out all 4 if so, but if there's one we don't support we should mention it.

  **Settings** > **Source Code Managers**.
* Added ability to manually create custom dependency exceptions under **Supply
  Chain** > **Settings**. This helps prevent blocking a pull request or merge
  request due to licensing issues. For example, if `bitwarden/cli@2023.9.0`,
  which has a GPL-3.0 license, is on the allowlist, setting a custom dependency
  exception means that the exclusion won't fail when upgrading to
  `bitwarden/cli@2023.9.1`.
* Projects are removed from SCP when the corresponding GitHub repository is
  archived on GitHub.
* **CLI tool**: 
  * Added color-coded severity icons, such as `❯❯❱`, to the CLI
  output for findings of known severity.

>   * Metrics sent from the CLI to Semgrep now include a breakdown of the number

Reviewer: Do you know which team added the metric in? I just checked our `privacy.md` and we don't mention it at all: https://github.com/semgrep/semgrep/blob/develop/PRIVACY.md#data-collected-as-metrics - I can follow up with them to have it added in.

Author: Yup, it's https://linear.app/semgrep/issue/PA-3312/add-findings-per-product-to-cli-scan-metrics (PM and/or Growth?)

---
slug: january-2024
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: January 2024
tags:
  - Release notes
---

# Semgrep release notes for January 2024

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in January 2024:
  * [<i class="fas fa-external-link fa-xs"></i>1.55.1](https://github.com/semgrep/semgrep/releases/tag/v1.55.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.55.2](https://github.com/semgrep/semgrep/releases/tag/v1.55.2)
  * [<i class="fas fa-external-link fa-xs"></i>1.56.0](https://github.com/semgrep/semgrep/releases/tag/v1.56.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.57.0](https://github.com/semgrep/semgrep/releases/tag/v1.57.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.58.0](https://github.com/semgrep/semgrep/releases/tag/v1.58.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.59.0](https://github.com/semgrep/semgrep/releases/tag/v1.59.0)

> ## 🌐 Cloud Platform

Reviewer: ```suggestion
## 🌐 Semgrep Cloud Platform
```

### Added

* Semgrep's Visual Studio Code extension now runs natively on Windows machines.

> * Added ability to test the connection to their source code manager by going to

Reviewer: ```suggestion
* Added ability to test the connection to their source code manager by going to
```
> ..."their" source code manager...
whose source code manager?

  **Settings** > **Source Code Managers**.

> * Added ability to manually create custom dependency exceptions under **Supply

Reviewer: Think this should be updated in our docs as well?

  Chain** > **Settings**. This helps prevent blocking a pull request or merge
  request due to licensing issues. For example, if `bitwarden/cli@2023.9.0`,
  which has a GPL-3.0 license, is on the allowlist, setting a custom dependency
  exception means that the exclusion won't fail when upgrading to
  `bitwarden/cli@2023.9.1`.
* Projects are removed from SCP when the corresponding GitHub repository is

>   archived.

Reviewer: ```suggestion
  archived on GitHub.
```

* **CLI tool**: 
  * Added color-coded severity icons, such as `❯❯❱`, to the CLI
  output for findings of known severity.

>   * Metrics sent from the CLI to Semgrep now include a breakdown of

Reviewer: Are these metrics for the user (in SCP dashboard) or for us, Semgrep Inc?

  the number of findings per product.

>   * Rules stored under a hidden directory, such as

Reviewer: I think this is worth mentioning in our scanning docs? Maybe the Ignore docs too actually.

  `dir/.hidden/myrule.yml`, are now processed when scanning with the `--config`
  flag.

### Changed

> * Improved the speed with which Semgrep Cloud Platform's **Dashboard** and

Reviewer: ```suggestion
- Improved loading times for **Dashboard** and **Findings** pages.
```
I'm using "loading" in this case because "load" could also mean "resource load"

>   **Findings** pages load.

Reviewer: ```suggestion
```
Delete if above suggestion is accepted.

* Renamed the **Upgrade** page to **Usage & billing**.

> * Redesigned the **Settings** > **Source Code Managers** page; changes include:

Reviewer: Lines 54 to 60 have variable tenses "displays" vs "renamed" - maybe pick a tense and stick with it?

  page, especially for organizations with many source code managers connected.
* The **Findings** page now displays issues present on multiple branches,
  regardless of which branch is used as a filter. 
* Temporarily removed support for writing rules using Jsonnet.

### Fixed

* Fixed an issue where the number of **Open** findings displayed in the
  navigation bar next to the **Code** link differed from the number displayed in
  the **Open** status filter.
* Fixed an issue where leaving a note on a finding also triaged the issue; users
  can now leave notes without changing the finding's triage state.
* Fixed an issue where paid subscribers couldn't submit support cases through
  the **Help** page.
* Fixed an issue where hovering over the Assistant's **Analyze** button caused
  the window to jitter.
* **CLI tool**: 
  * Fixed an issue where multi-line comments in Dockerfiles weren't
  parsed correctly.
  * Fixed an issue where Semgrep used `/tmp` instead of the path set
    in the `TMPDIR` environment variable for the Semgrep cache.
  * Fixed an issue where Semgrep would error on reading a
    `nosemgrep` comment with multiple rule IDs.

>   * Fixed a bug where Gemfile.lock files with multiple `GEM`

Reviewer: ```suggestion
  * Fixed a bug where `Gemfile.lock` files with multiple `GEM`
```

toc_max_heading_level: 3
title: January 2024
tags:
  - Release notes
---

# Semgrep release notes for January 2024

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in January 2024:
  * [<i class="fas fa-external-link fa-xs"></i>1.55.1](https://github.com/semgrep/semgrep/releases/tag/v1.55.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.55.2](https://github.com/semgrep/semgrep/releases/tag/v1.55.2)
  * [<i class="fas fa-external-link fa-xs"></i>1.56.0](https://github.com/semgrep/semgrep/releases/tag/v1.56.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.57.0](https://github.com/semgrep/semgrep/releases/tag/v1.57.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.58.0](https://github.com/semgrep/semgrep/releases/tag/v1.58.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.59.0](https://github.com/semgrep/semgrep/releases/tag/v1.59.0)

## 🌐 Cloud Platform

### Added

* Semgrep's Visual Studio Code extension now runs natively on Windows machines.
* Added ability for organizations to test connections to GitHub and GitLab by going to
  **Settings** > **Source Code Managers**.

> * Projects are removed from SCP when the corresponding GitHub repository is

Reviewer 2: Not sure how you want to handle this but technically the projects will be visible on the not-scanning part of the SCP

Author: @annikalee Do you think something like this would be better?

"Projects are moved from the **Scanning** to **Not scanning** tab when the corresponding GitHub repository is archived"

Reviewer 2: Yeah that sounds clearer

  archived on GitHub.
* **CLI tool**: 
  * Added color-coded severity icons, such as `❯❯❱`, to the CLI
  output for findings of known severity.
  * Metrics sent from the CLI to Semgrep now include a breakdown of the number

>   of findings per product. This helps the Semgrep team understand users better.

Reviewer 2: I'm not sure what this means. But I think this was a growth project so maybe @bmahe does?

Author: This was [changed in release 1.56.0](https://github.com/semgrep/semgrep/releases/tag/v1.56.0).

Do you mean that the description is too vague?

Reviewer 2: Yeah but it looks like it matches what we said publicly so maybe it's okay. I just didn't know what "This helps the Semgrep team understand users better." really meant :p

  * Rules stored under a hidden directory, such as
  `dir/.hidden/myrule.yml`, are now processed when scanning with the `--config`
  flag.

### Changed

* Renamed the **Upgrade** page to **Usage & billing**.
* Redesigned the **Settings** > **Source Code Managers** page; changes include:
  * Renamed the **Remove SCM config** button to **Disconnect**.
  * Set the **Remove app** button to only show up for registered GitHub apps.
* Improved the page load times for the **Settings** > **Source Code Managers**
  page, especially for organizations with many source code managers connected.

> * Temporarily removed support for writing rules using Jsonnet.

Reviewer 2: Should we include this?

  `[dev-packages]` section.
* Fixed a bug where `Gemfile.lock` files with multiple `GEM` sections weren't parsed correctly.

## 🔐 Secrets (beta)

### Fixed

- Fixed a bug with custom secrets rules in which rule visibility could be set to `unlisted`. Now, To protect the privacy of secrets rules, users cannot set Secrets rules to any other visibility except for **private**. <!-- 12039, 12040, 12025 -->

## 📝 Documentation and knowledge base

### Added

- Added [legal information](/semgrep-code/semgrep-assistant-code/#privacy-and-legal-considerations) about Semgrep Assistant. <!-- 1308 -->
- Added documentation about Semgrep Assistant's Component and Recommendation filters. <!-- 1324 -->
- Knowledge base articles:
    - Troubleshoot why [SAML stops working](/kb/semgrep-cloud-platform/saml-stops-working/) <!-- 1330 -->
    - [Troubleshooting "You are seeing this because the engine was killed" on monorepos](/kb/semgrep-code/scan-engine-kill/) <!-- 1310 -->
- Added guidance on running Semgrep Supply Chain scans [in the CLI](/semgrep-supply-chain/getting-started/#run-a-scan-using-the-cli ). <!-- 1334 -->

### Changed

> - Updated API docs to use the term `teams`. The use of the term `groups` is deprecated.

Reviewer 2: This is for a private beta feature. I think we can remove it

Reviewer: Removed!
